token/js/src/extensions/extensionType.ts

```diff
@@ -176,17 +185,32 @@ export function getAccountTypeOfMintType(e: ExtensionType): ExtensionType {
     }
 }
 
-function getLen(extensionTypes: ExtensionType[], baseSize: number): number {
+function getLen(
+    extensionTypes: ExtensionType[],
+    baseSize: number,
+    variableLengthExtensions?: [ExtensionType, number][]
+): number {
     if (extensionTypes.length === 0) {
         return baseSize;
     } else {
-        const accountLength =
+        let accountLength =
             ACCOUNT_SIZE +
             ACCOUNT_TYPE_SIZE +
             extensionTypes
                 .filter((element, i) => i === extensionTypes.indexOf(element))
                 .map((element) => addTypeAndLengthToLen(getTypeLen(element)))
                 .reduce((a, b) => a + b);
+        if (variableLengthExtensions) {
```

Comment:
nit / up to you: if the parameter `variableLengthExtensions` becomes a default value `= {}` rather than a nullable, then you can remove the if check and just add the next filter, which would remove this pesky branch